src/test/java/org/eclipse/yasson/internal/AnnotationIntrospectorWithoutJavaDektopModuleTest.java

```diff
@@ -0,0 +1,47 @@
+package org.eclipse.yasson.internal;
+
+import static org.junit.Assert.assertNull;
+
+import org.eclipse.yasson.internal.AnnotationIntrospectorTestFixtures.ObjectWithMissingConstructorAnnotation;
+
+import javax.json.bind.JsonbConfig;
+
+import javax.json.spi.JsonProvider;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Tests the {@link AnnotationIntrospector} with missing optional module "java.deskop", <br>
+ * that contains the ConstructorProperties-Annotation.
+ * <p>
+ * Requires --limit-modules java.base,java.logging,java.sql (to exclude java.desktop) to work. <br>
+ * See pom.xml surefire plugin configuration.
+ * 
+ * @see AnnotationIntrospectorTestFixtures
+ */
+public class AnnotationIntrospectorWithoutJavaDektopModuleTest {
+
+    private JsonbContext jsonbContext = new JsonbContext(new JsonbConfig(), JsonProvider.provider());
+
+    /**
+     * class under test.
+     */
+    private AnnotationIntrospector instrospector = new AnnotationIntrospector(jsonbContext);
+
+    @Test
+    public void testNoConstructorPropertiesAnnotationWithoutModuleJavaDesktop() {
+        String className = "java.beans.ConstructorProperties";
+        try {
+            Class.forName(className);
+            Assert.fail("Class [" + className + "] should not be available for this test.");
+        } catch (ClassNotFoundException e) {
+            // OK, as expected
+        }
+    }
+
+    @Test
+    public void testCreatorShouldBeNullOnMissingConstructorAnnotation() {
```

Comment:
Furthermore we should test that Creator is not null when `@JsonbCreator` is used without `java.naming` module. This may seem obvious, but after further future refactoring it may stop working at some point and the test should make sure it doesn't.

Reply:
I renamed the test to "AnnotationIntrospectorWithoutOptionalModulesTest" and added two more test cases to check, if JsonbCreator still works on constructors and static factory methods.